app/controllers/account_activations_controller.rb

```diff
@@ -0,0 +1,15 @@
+class AccountActivationsController < ApplicationController
+  def edit
+    user = User.find_by email: params[:email]
+
+    if user && !user.activated? && user.authenticated?(:activation, params[:id])
```

Comment:
1. Sử dụng The Safe Navigation Operator (&.) chỗ này được k em?
2. params[:id] thì liên quan gì mà truyền vào method authenticated? em

Reply:
1. E nghĩ là không được ạ. Vì trong trường hợp :activation or params[:id] bị false thì nó sẽ bắn ra exception.
2. params[:id] là 1 phần để tạo ra token , token này dùng để so sánh với token trong attribute của user xem có đúng là user đó hay không.